### ISSUE TITLE
Video Series on .NET Library Concepts

### ISSUE THREAD

**Issue by terrajobst** [terrajobst/videos#7]
It would be worthwhile to go over the key areas I covered in my [recent .NET Standard talk](https://channel9.msdn.com/Events/dotnetConf/2018/S107) and have dedicated videos on:

* Versioning
* Strong Naming
* Multi-targeting
    - Understanding rules when multi-targeting:
        + .NET Standard output must be a subset
* Building packages
* SourceLink
* Nerdbank.GitVersioning
* Automatic Binding Redirects
* MSBuild
    - Autogenerated `AssemblyInfo.cs`
    - `Directory.Build.props`/`Directory.Build.targets`

**Comment by cidico** [comment]
May I add that would be really nice to see common scenarios to solve with Directory.Build.props/Directory.Build.targets?

**Comment by terrajobst** [comment]
Yep, it's in my list at the bottom. I think this video will also need some rudimentary intro to MSBuild and how the projects are built so that people understand the difference between `.props` and `.targets`. The examples I had in mind were:

* Centralizing common metadata (like NuGet authorship and versioning)
* Centralizing configuration policies (such as enabling strong naming for all library projects)
* Centralizing version numbers of package dependencies
* Automatically adding packages based on the project's role (such as adding `xunit` to all projects that end with `.Tests`).

@cidico, do you have other specifics in mind?

**Comment by cidico** [comment]
I saw your video @ .net conf about adding automagically the nugets. **VERY INTERESTING**!!! 
Right now I don't have any fancy scenario in mind...
Maybe the "common poor man job" that most enterprise devs have.. Like...

- Updating the database at build time (based on configuration, of course...);

I once worked with a solution that had a very nasty build process.
It needed to compile some C++ projects and copy the result .dlls to a folder and then build another project and them copy to another...
Some dlls needed to be registred on Windows during the build...
